### PR TITLE
Search path for ffmpeg before downloading

### PR DESCRIPTION
**Pull request by tdsmith**
Brings behavior of get_exe in line with FfmpegFormat docstring:

    Otherwise, imageio will look for ffmpeg on the system path and then
    download ffmpeg if not found.

Also cache the result of get_exe on the classes, because shelling out to ffmpeg -version is probably too expensive to do on each instance.

### CODE REVIEW

File: imageio/plugins/ffmpeg.py [hunk 1]
```diff
@@ -45,6 +45,12 @@ def get_exe():
     if exe:  # pragma: no cover
         return exe
 
+    try:
+        sp.check_output(["ffmpeg", "-version"])
```
Comment (tdsmith) [inline imageio/plugins/ffmpeg.py]: `check_output` doesn't exist on 2.6.

File: imageio/plugins/ffmpeg.py [hunk 2]
```diff
@@ -45,6 +45,12 @@ def get_exe():
...
+        return "ffmpeg"
+    except sp.CalledProcessError:
```
Comment (tdsmith) [inline imageio/plugins/ffmpeg.py]: `FileNotFoundError` is raised on 3.x (and doesn't exist on 2.x), which inherits from `OSError`.

File: imageio/plugins/ffmpeg.py [hunk 3]
```diff
@@ -45,6 +45,15 @@ def get_exe():
     if exe:  # pragma: no cover
         return exe
 
+    # Check if ffmpeg is in PATH
+    try:
+        sp.check_call(["ffmpeg", "-version"], stdout=sp.PIPE, stderr=sp.PIPE)
```
Comment (tdsmith) [inline imageio/plugins/ffmpeg.py]: Nope, this can deadlock.

**Comment by almarklein** [comment]
LGTM. Will merge in a day or two unless @kuchi has comments.

**Comment by kuchi** [comment]
LGTM, all tests pass on my OS X.

**Comment by almarklein** [comment]
Thanks both!

**Comment by tdsmith** [comment]
Thanks for a great tool and a quick response; very gratifying :)

**Pull request merged**